src/argilla/client/feedback/dataset/base.py

```diff
@@ -230,45 +176,18 @@ def __init__(
                 raise ValueError(
                     "Expected `guidelines` to be either None (default) or a non-empty string, minimum length is 1."
                 )
-        self.__guidelines = guidelines
-
-        self._records: List[FeedbackRecord] = []
-        self._new_records: List[FeedbackRecord] = []
-
-    def __len__(self) -> int:
-        """Returns the number of records in the dataset."""
-        return len(self.records)
-
-    def __getitem__(self, key: Union[slice, int]) -> Union[FeedbackRecord, List[FeedbackRecord]]:
-        """Returns the record(s) at the given index(es).
+        self._guidelines = guidelines
 
-        Args:
-            key: the index(es) of the record(s) to return. Can either be a single index or a slice.
-
-        Returns:
-            Either the record of the given index, or a list with the records at the given indexes.
-        """
-        if len(self.records) < 1:
-            raise RuntimeError(
-                "In order to get items from `rg.FeedbackDataset` you need to either add"
-                " them first with `add_records` or fetch them from Argilla or"
-                " HuggingFace with `fetch_records`."
-            )
-        if isinstance(key, int) and len(self.records) < key:
-            raise IndexError(f"This dataset contains {len(self)} records, so index {key} is out of range.")
-        return self.records[key]
+    @property
+    @abstractproperty
+    def records(self) -> Any:
```

Comment:
Could be expected an `Iterable[FeedbackRecord]` here?

Reply:
No because `records` is not an iterable, it's a `List[FeedbackRecord]` in `FeedbackDataset` while that's an instance of `RemoteFeedbackRecords` in `RemoteFeedbackDataset`